src/overlays/actors/ovl_Bg_Ctower_Gear/z_bg_ctower_gear.c

```diff
@@ -21,18 +39,189 @@ const ActorInit Bg_Ctower_Gear_InitVars = {
     (ActorFunc)BgCtowerGear_Update,
     (ActorFunc)BgCtowerGear_Draw
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Ctower_Gear_0x80AD2B70/func_80AD2B70.asm")
+extern Gfx D_06010828[];
+
+extern Gfx D_06015F30[];
+
+extern Gfx D_060160A0[];
+
+extern CollisionHeader D_06016E70;
+
+extern Gfx D_06017018[];
+
+extern Gfx D_06018118[];
+
+extern CollisionHeader D_06018588;
+
+static Vec3f D_80AD3270[] = {
+    {-70.0f, -60.0f, 8.0f},
+    {-60.0f, -60.0f, -9.1f},
+    {-75.0f, -60.0f, -9.1f},
+    {-70.0f, -60.0f, -26.2f}
+};
+
+static Vec3f D_80AD32A0[] = {
+    {85.0f, -60.0f, 8.0f},
+    {85.0f, -60.0f, -9.1f},
+    {85.0f, -60.0f, -26.2f},
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 400, ICHAIN_STOP),
+};
+
+static InitChainEntry sInitChainCenterCog[] = {
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 1500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 2000, ICHAIN_STOP),
+};
+
+static InitChainEntry sInitChainOrgan[] = {
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 420, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 570, ICHAIN_STOP),
+};
+
+static Gfx* D_80AD32E8[] = {D_06010828, D_06017018, D_06018118};
+
+void BgCtowerGear_Splash(BgCtowerGear *this, GlobalContext *GlobalContext) {
+    int i;
+    s32 sp88;
+    Vec3f sp7C;
+    Vec3f sp70;
+    s32 pad;
+    int j;
+    s16 sp66;
+
+    sp88 = this->dyna.actor.flags & 0x40;
+    sp66 = this->dyna.actor.shape.rot.z & 0x1FFF;
+    if ((sp88 != 0) && (sp66 < 0x1B58) && (sp66 >= 0x1388)) {
+        Matrix_RotateY(this->dyna.actor.home.rot.y, 0);
+        SysMatrix_InsertXRotation_s(this->dyna.actor.home.rot.x, 1);
+        SysMatrix_InsertZRotation_s(this->dyna.actor.home.rot.z, 1);
+        for(i=0; i<4; i++) {
+            if ((u32) Rand_Next() >= 0x40000000) {
+                sp70.x = D_80AD3270[i].x - (Rand_ZeroOne() * 30.0f);
+                sp70.y = D_80AD3270[i].y;
+                sp70.z = D_80AD3270[i].z;
+                SysMatrix_MultiplyVector3fByState(&sp70, &sp7C);
+                sp7C.x += this->dyna.actor.world.pos.x + ((Rand_ZeroOne() * 20.0f) - 10.0f);
+                sp7C.y += this->dyna.actor.world.pos.y;
+                sp7C.z += this->dyna.actor.world.pos.z + ((Rand_ZeroOne() * 20.0f) - 10.0f);
+                EffectSsGSplash_Spawn(GlobalContext, &sp7C, NULL, NULL, 0, ((u32) Rand_Next() >> 25) + 340);
```

Comment:
Based on the context of `EffectSsGSplash_Spawn`,  sp7C is a position vector, perhaps the position of the splash. sp70 could be the starting position of the cogs, I am not positive on that though.

Reply:
sp7C makes sense to be splash spawn pos, but still not sure on sp70. I don't think it is the starting position of the cog it get initialized to different values all in the same frame. Not to mention the x coordinate is randomized as well.